AmplifyPlugins/Core/AWSPluginsCore/Model/Support/AuthRule+Extension.swift

```diff
@@ -7,6 +7,43 @@
 
 import Amplify
 
+protocol AWSAuthRuleProvider {
+    static var iam: AuthRuleProvider { get }
+    static var userPools: AuthRuleProvider { get }
+}
+
+extension AuthRuleProvider: AWSAuthRuleProvider {
+    private static func valueForIAM() -> String { "iam" }
+    private static func valueForCognitoUserPool() -> String { "userPools" }
+
+    public static var iam: AuthRuleProvider {
+        .custom(name: valueForIAM())
+    }
+
+    public static var userPools: AuthRuleProvider {
+        .custom(name: valueForCognitoUserPool())
+    }
+
+    public func toAWSAuthorizationType() throws -> AWSAuthorizationType {
```

Comment:
what is the difference between `AWSAuthorizationType` and `AuthRuleProvider`? Is it the latter constrained to the metadata and the former used in the auth implementation?

Reply:
Yes, that's correct. `AuthRuleProvider` defines values allowed by `@auth` rule specs